src/components/Redux/RocketsSlice/RocketSlice.js

```diff
@@ -1,13 +1,37 @@
-import { createSlice } from '@reduxjs/toolkit';
-import data from '../../TempRocketData';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+/* import data from '../../TempRocketData'; */
+
+const url = 'https://api.spacexdata.com/v3/rockets';
 
 const initialState = {
-  RocketDatile: data,
+  RocketDatile: [],
+  loading: false,
+  error: null,
 };
 
+export const rocketItems = createAsyncThunk('rocket', async () => {
+  const response = await fetch(url);
+  const result = response.json();
+  return result;
+});
+
 const rocketSlice = createSlice({
   name: 'rocket',
   initialState,
+  extraReducers: {
+    [rocketItems.pending]: (state) => {
+      state.loading = true;
```

Comment:
Great work on adding extraReducers to handle the promise made on AsyncThunk
In order that we don't lose the content from "state". You need to call the state with "...state" and then modify the desired attribute in this case loading.